addon/components/discover-page/component.js

```diff
@@ -0,0 +1,808 @@
+            } else if (key === 'contributors') {
+                queryKey = 'lists.contributors.name';
+            }
+
+            query[queryKey] = lockedParams[key];
+            queryBody.push({
+                term: query
+            });
+        });
+        return queryBody;
+    },
+    getCounts() {
+        // Ember-SHARE method
+        let queryBody = JSON.stringify({
+            size: 0,
+            aggregations: {
+                sources: {
+                    cardinality: {
+                        field: 'sources',
+                        precision_threshold: MAX_SOURCES
+                    }
+                }
+            }
+        });
+        return Ember.$.ajax({
+            url: this.get('searchUrl'),
+            crossDomain: true,
+            type: 'POST',
+            contentType: 'application/json',
+            data: queryBody
+        }).then((json) => {
+            this.setProperties({
+                numberOfEvents: json.hits.total,
+                numberOfSources: json.aggregations.sources.value
+            });
+        });
+    },
+    getQueryBody() {
+        /**
+         * Builds query body to send to SHARE from a combination of locked Params, facetFilters and activeFilters
+         *
+         * @method getQueryBody
+         * @return queryBody
+         */
+        let filters = this.buildLockedQueryBody(this.get('lockedParams')); // Empty list if no locked query parameters
+        // From Ember-SHARE. Looks at facetFilters (partial SHARE queries already built) and adds them to query body
+        let facetFilters = this.get('facetFilters');
+        for (let k of Object.keys(facetFilters)) {
+            let filter = facetFilters[k];
+            if (filter) {
+                if (Ember.$.isArray(filter)) {
+                    filters = filters.concat(filter);
+                } else {
+                    filters.push(filter);
+                }
+            }
+        }
+
+        // For PREPRINTS and REGISTRIES.  Adds activeFilters to query body.
+        const activeFilters = this.get('activeFilters');
+        const filterMap = this.get('filterMap');
+        for (const key in filterMap) {
+            const val = filterMap[key];
+            const filterList = activeFilters[key];
+
+            if (!filterList.length || (key === 'providers' && this.get('theme.isProvider')))
+                continue;
+
+            filters.push({
+                terms: {
+                    [val]: filterList
+                }
+            });
+        }
+
+        // For PREPRINTS and REGISTRIES. If theme.isProvider, add this provider to the query body
+        if (this.get('theme.isProvider') && this.get('providerName') !== null) {
+            filters.push({
+                terms: {
+                    sources: [this.get('providerName')]
+                }
+            });
+        }
+
+        let query = {
+            query_string: {
+                query: this.get('q') || '*'
```

Comment:
This is how Ember-SHARE and preprints are currently writing this. I think this is so a blank text field can query.